### PR TITLE
Portal page authentication

### PR DESCRIPTION
**Pull request by MohammedIrheem**
** Portal page authentication **
  - validate a pinCode by using Yup
  - compare the pinCode with the hashed pinCode from the database by using bcrypt
  - generate a token to make protected routes

### CODE REVIEW

File: server/router/index.js [hunk 1]
```diff
@@ -53,13 +61,19 @@ router.post(
 router.post('/cancelUser', validationCancelReg, cancelRegistration);
 router.post('/login', loginValidation, checkEmailIfExist, checkPassword, login);
 
-// should be protected
-router.use(protectedRoute);
+
+router.post('/portal/login', pinCodeValidation, getHashedPinCode, checkPinCode);
+// pin code protection for portal route
+router.use(protectedPortalRoute);
 router.post('/event', validateEvent, createEvent);
```
Comment (AlaaTaima) [inline server/router/index.js]: this route is protected from `protectedRoute` middle ware 
```suggestion
router.use(protectedRoute);
router.post('/event', validateEvent, createEvent);
```

File: server/controller/routes/admin/portalLogainPage.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+const getHashedPinCode = (req, res, next) => {
+  getAdmin().then(({ rows }) => {
+    const [admin] = rows;
+    req.admin.pinCode = admin.pinCode;
+    next();
+  });
+};
+
+module.exports = { pinCodeValidation, getHashedPinCode };
```
Comment (AlaaTaima) [inline server/controller/routes/admin/portalLogainPage.js]: this file relates to portal so, it should be at portal folder

File: server/controller/routes/admin/portalLogainPage.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    })
+    .catch((err) => {
+      res.status(400).json({ msg: err.message });
+    });
+};
+
+const getHashedPinCode = (req, res, next) => {
+  getAdmin().then(({ rows }) => {
+    const [admin] = rows;
+    req.admin.pinCode = admin.pinCode;
```
Comment (AlaaTaima) [inline server/controller/routes/admin/portalLogainPage.js]: fix it : 
`    req.pinCode = admin.pinCode;
`

File: server/middleware/checkPinCode.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      const err = new Error();
+      err.msg = 'incorect pinCode';
+      err.status = 401;
+      next(err);
+    } else {
+      const token = jwt.sign({ id: req.admin.id }, process.env.SECRET_KEY_PORTAL);
+      res.cookie('portalToken', token);
+      // next();
+      res.send('Hi');
+    }
```
Comment (AhmedSafi97) [inline server/middleware/checkPinCode.js]: please remove this line `  // next();` and send the response in JSON format with more descriptive message

**Comment by AhmedSafi97** [comment]
You did a really good job :+1: 
keep up the hard work  :muscle:

**Pull request merged**